lib/bloc/location/location_event.dart

```diff
@@ -2,6 +2,11 @@ part of 'location_bloc.dart';
 
 abstract class LocationEvent {}
 
+
+class RequestPermissionLocationEvent extends LocationEvent {}
+
 class RequestCurrentLocationEvent extends LocationEvent {}
 
 class RequestLocationServiceEvent extends LocationEvent {}
+
+class DefaultLocationServiceEvent extends LocationEvent {}
```

Comment:
all logic can be within `location_bloc`, so there's no need to have `DefaultLocationServiceEvent`